extensions/agroal/deployment/src/main/resources/dev-ui/qwc-agroal-datasource.js

```diff
@@ -0,0 +1,339 @@
+                        @value-changed="${this._onDataSourceChanged}"
+                        .allowCustomValue="${false}"
+                    ></vaadin-combo-box>
+                `;
+    }
+    
+    _renderSelectedDatasource(){
+        if(this._selectedDataSource){
+            return html`<code>${this._selectedDataSource.jdbcUrl}</code>`;
+        }
+    }
+    
+    _renderTables(){
+        if(this._tables){
+            
+            return html`<qui-card class="tablesCard" header="Tables">
+                            <div slot="content">
+                                <vaadin-list-box selected="0" @selected-changed="${this._onTableChanged}">
+                                    ${this._tables.map((table) =>
+                                        html`<vaadin-item>${table.tableName}</vaadin-item>`
+                                    )}
+                                </vaadin-list-box>
+                            </div>
+                        </qui-card>`;
+        }
+        
+    }
+    
+    _renderDataAndDefinition(){
+        return html`<vaadin-tabsheet class="fill" theme="bordered">
+                        <vaadin-button slot="suffix" theme="icon" title="Stop Refresh" aria-label="Refresh">
+                            <vaadin-icon @click=${this.hotReload} icon="font-awesome-solid:arrows-rotate"></vaadin-icon>
+                        </vaadin-button>
+                        
+                        ${this._renderWatchButton()}
+                        
+                        <vaadin-tabs slot="tabs">
+                          <vaadin-tab id="data-tab">Data</vaadin-tab>
+                          <vaadin-tab id="definition-tab">Definition</vaadin-tab>
+                        </vaadin-tabs>
+
+                        <div tab="data-tab" style="height:100%;">${this._renderTableData()}</div>
+                        <div tab="definition-tab" style="height:100%;">${this._renderTableDefinition()}</div>
+                      </vaadin-tabsheet>`;
+    }
+    
+    _renderWatchButton(){
+        if(this._isWatching){
+            return html`<vaadin-button slot="suffix" theme="icon" title="Stop watching" aria-label="Stop watching">
+                            <vaadin-icon @click=${this._unwatch} icon="font-awesome-solid:eye"></vaadin-icon>
+                        </vaadin-button>`;
+        }else{
+            return html`<vaadin-button slot="suffix" theme="icon" title="Watch" aria-label="Watch">
+                            <vaadin-icon @click=${this._watch} icon="font-awesome-solid:eye-slash"></vaadin-icon>
+                        </vaadin-button>`;
+        }
+    }
+    
+    _renderTableData(){
+        if(this._selectedTable && this._currentDataSet && this._currentDataSet.cols){
+            return html`<div class="data">
+                            <vaadin-grid .items="${this._currentDataSet.data}" theme="row-stripes no-border" class="fill" column-reordering-allowed>
```

Comment:
I'm wondering what happens if you have a `bytea` in your table? Or some text that is very large.

Reply:
I have a lot of `byte[]` in my entities :) Images, mostly.